pages/Logger.md

```diff
@@ -45,27 +97,144 @@ enum LogLevel {
 
 ## API
 
-There are two main methods:
+```kotlin
+// Kotlin
+
+  /** Logs a String object using a given level.  */
+  fun log(level: Level, tag: String? = null, message: String)
+
+  /** Logs a String and Throwable object using a given level.  */
+  fun log(level: Level, throwable: Throwable, tag: String? = null, message: String)
+
+  /** Logs a key-value pair.  */
+  fun log(key: String, value: Any?)
+
+  /** Method to send issue to external servers or local files.  */
+  funendIssue(tag: String, message: String)
+
+  /** Sets a device detail with boolean type.  */
+  fun setDeviceBoolean(key: String, value: Boolean)
+
+  /** Sets a device detail with string type.  */
+  fun setDeviceString(key: String, value: String)
+
+  /** Sets a device detail with integer type.  */
+  funetDeviceInteger(key: String, value: Int)
+
+  /** Sets a device detail with double type.  */
+  fun setDeviceFloat(key: String, value: Float)
+
+  /** Remove a device detail.  */
+  fun removeDeviceKey(key: String)
 
-```TypeScript
+  /** Get the device identifier generated.  */
+  val deviceIdentifier: String
+```
+
+```swift
+// Swift
+
+/// Abstracts concrete implementations of a logger system.
+@objc(MJLogger) public protocol Logger {
+    
+    /// Logs a String object using a given level
+    ///
+    /// - Parameters:
+    ///   - level: Type of log.
+    ///   - tag: An additional label to help categorise logs.
+    ///   - message: The message to be logged.
+    func log(level: LogLevel, tag: String?, message: String)
+    
+    /// Logs a key-value pair
+    ///
+    /// - Parameters:
+    ///   - key: They key
+    ///   - value: The value
+    func log(key: String, value: Any?)
+}
+```
+
+```typescript
 // TypeScript
+
 interface Logger {
     logKeyValue(key: string, value: any): void;
 
     log(level: LogLevel, message: string): void;
     log(level: LogLevel, tag: string, message: string): void;
 }
-```
```

Comment:
We should keep this closing code block fence (```) here

Reply:
done